### DISCUSSION TITLE
Setting :key on a component doesn't seem to work

### DISCUSSION THREAD

**Discussion by mathiasgrimm** [livewire/livewire#8086] [7. Bugs (archived)]
### Livewire version

v3.4.7

### Laravel version

v10.47.0

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

1. Create a full page component
2. Create a non full page component
3. Include this non-full page component and set `:key=""` or key="" within a loop (@foreach for example)
4. Html doesn't have the defined id but has the random(20) instead

### Repository or Wirebox URL

https://wirebox.app/b/4z1gz

### Additional information

[The code has been like that for a long time ](https://github.com/livewire/livewire/blob/0335b8f022ac535fc3cf29fd8a8fcb275353a470/src/Mechanisms/HandleComponents/HandleComponents.php#L41 
)

However the [documentation](https://livewire.laravel.com/docs/nesting#rendering-children-in-a-loop) instructs use to use `:key` when rendering children in a loop

```php
<div>
    <h1>Todos</h1>
 
    @foreach ($todos as $todo)
        <livewire:todo-item :$todo :key="$todo->id" />
    @endforeach
</div>
```

<img width="1209" alt="image" src="https://github.com/livewire/livewire/assets/450069/cde5f033-e7b3-4586-984a-5306a9f9ee58">

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by mathiasgrimm** [comment]
Anybody?

**Comment by danie-ramdhani** [comment]
i don't think that how livewire `:key` in livewire nesting works... if you want to set your component custom id, use `$this->setId(...)`

https://wirebox.app/b/xp3el

**Reply by mathiasgrimm** [reply]
https://livewire.laravel.com/docs/nesting#rendering-children-in-a-loop

**Comment by joshhanley** [comment]
@mathiasgrimm a `:key/wire:key` on a component doesn't get rendered into the HTML but it is tracked internally inside the component. The `wire:id` is not the same thing as `wire:key`.
